app/scripts/lib/AccountIdentitiesPetnamesBridge.ts

```diff
@@ -0,0 +1,52 @@
+import { NameController, NameType } from '@metamask/name-controller';
+import {
+  PreferencesController,
+  AccountIdentityEntry,
+} from '../controllers/preferences';
+import {
+  PetnameEntry,
+  AbstractPetnamesBridge,
+  PetnamesBridgeMessenger,
+} from './AbstractPetnamesBridge';
+import { FALLBACK_VARIATION } from 'ui/hooks/useName';
+
+/**
+ * A petnames bridge that uses the account identities from the preferences controller as the source.
+ */
+export class AccountIdentitiesPetnamesBridge extends AbstractPetnamesBridge {
+  #preferencesController: PreferencesController;
+
+  constructor({
+    preferencesController,
+    nameController,
+    messenger,
+  }: {
+    preferencesController: PreferencesController;
+    nameController: NameController;
+    messenger: PetnamesBridgeMessenger;
+  }) {
+    super({ isTwoWay: false, nameController, messenger });
+    this.#preferencesController = preferencesController;
+  }
+
+  /**
+   * @override
+   */
+  protected getSourceEntries(): PetnameEntry[] {
+    const { identities } = this.#preferencesController.store.getState();
+    return Object.values(identities).map((identity: AccountIdentityEntry) => ({
+      value: identity.address,
```

Comment:
Do we need to normalize these addresses?

Are they always stored in lowercase or the checksummed address like the address book?

Reply:
I checked it out, and here's what I found: address normalization is already taken care of by the name controller when setting a name [1], and by the useName hook when reading a name [2]. Looks like we're all set on this front, so no extra steps needed here.

[1] https://github.com/MetaMask/core/blob/be55cebdc84cb7d7f897ce7f8bfe0279bdecfdc1/packages/name-controller/src/NameController.ts#L383-L384
[2] https://github.com/MetaMask/metamask-extension/blob/8b89c3e7dbfd9d5b7317e73dc84ff2000f5b7fac/ui/hooks/useName.ts#L8-L9

Reply:
Sorry for the confusion, I'm referring to the case of the account identity addresses.

We compare them in `#computeChangeList` using their keys, so do we want to ensure the source entries are lowercase since the petnames addresses will be lowercase?

Reply:
Just highlighting this comment, thanks.

Reply:
Great catch, Matt. Given that this detail can be overlooked (as I just did), would it make sense to handle the lower-casing in the AbstractPetnameBridge, (in "getKey", for example)?

Reply:
Absolutely, good idea!

Reply:
Done.

Reply:
Done.